devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/dependency/ApplicationDeploymentClasspathBuilder.java

```diff
@@ -213,7 +214,9 @@ private void setUpDeploymentConfiguration() {
                     final DependencyHandler dependencies = project.getDependencies();
                     final Set<Dependency> deploymentDependencies = new HashSet<>();
                     for (ExtensionDependency extension : extensions) {
-                        if (extension instanceof LocalExtensionDependency) {
+                        if (extension instanceof IncludedBuildExtensionDependency) {
+                            deploymentDependencies.add(((IncludedBuildExtensionDependency) extension).getDeployment());
```

Comment:
Could we do it the way it's done below? i.e.
````
deploymentDependencies.add(dependencies.create(
                                    extension.getDeploymentModule().getGroupId() + ":"
                                            + extension.getDeploymentModule().getArtifactId() + ":"
                                            + extension.getDeploymentModule().getVersion()));
````

Reply:
It might work, as it creates a similar object (the one returned from the extension object is an ExternalModuleDependecy while the one returned from the dependencies.create is a decorated object), in any case we would need to differentiate between a locally built extension and a composite built one. I would rather consolidate these 3 type into a self describing one, but didn't want to change the current code to that extent. I also dislike the addition of the boolean parameter to the loadExtensionInfo in the ToolingUtils class, but again, didn't want to change the code to that extent.